### ISSUE TITLE
The `never` option of `declaration-block-trailing-semicolon` takes away the semicolon Less asks for behind a declaration that spells no value

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#358]
Less asks for the semicolon behind a declaration that spells no value, and the `never` option of `declaration-block-trailing-semicolon` takes it away:

```less
a {
	color:;
}
```

comes back from `--fix` as

```less
a {
	color:
}
```

and `less.render` compiles the first and refuses the second with `Unrecognised input. Possibly missing opening '{'`. Nothing is reported on the second run, since the rule got what it asked for. `a { --x:; }` goes the same way. This is [#309](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/309) on another node class: there the semicolon closed an at-rule, here it closes a declaration whose value is missing, and Less reads the block as unclosed either way.

A value is missing to Less wherever nothing but whitespace and comments stands behind the colon. Measured against `less@4.9.0`, the whole matrix:

| Last node of the block | Less without the semicolon |
| --- | --- |
| `color:` | refused |
| `color: ` | refused |
| `color: /*c*/` | refused |
| `color: !important` | refused |
| `color: pink` | compiles |
| `color: pink !important` | compiles |
| `--x:` | refused |
| `--x: ` | refused |
| `--x: /*c*/` | refused |
| `--x: !important` | **compiles** |
| `--x: pink` | compiles |

The two node classes part on the flag. An ordinary declaration spells no value in `!important`, and Less refuses `a { color: !important }`; a custom property takes everything behind the colon literally, so the flag is its value and `a { --x: !important }` compiles. `postcss-less` files the flag in `raws.important` for both, so the reading has to ask which kind of property it is rather than read one text.

An at-rule the parser hands over as a Less variable declaration goes with the ordinary declaration: `a { @v: !important }` is refused, and that half is answered already by `requiresTrailingSemicolon`, added for #309.

## Why no oracle sees it

`converge.mjs` parses the output with the syntax that wrote it, and `postcss-less` reads `color:` without a semicolon perfectly well. The parser is not the compiler, and only the compiler minds — the same reason [#309](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/309) and [#231](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/231) name.

## What would end it

`requiresTrailingSemicolon` carries the at-rule half and turns a declaration away at its first line. Giving it the declaration half means one more reading: the value the file spells, comments blanked, the important flag not counting where the property is not a custom property. The option cannot be satisfied for such a declaration at all, so the fix has to be declined and the warning left standing, exactly as it is for an at-rule.

Empty custom property values are a shape this project already recognises — `--custom-prop: ;` and `--custom-prop:;` were given an exception in `declaration-block-semicolon-space-before` in 5.2.0 — so the form is not hypothetical.

Found while writing the branch for #309, where the matrix above was measured.
